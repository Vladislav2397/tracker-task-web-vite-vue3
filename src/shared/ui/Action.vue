<template>
    <div :class="classes"></div>
</template>

<script lang="ts" setup>
import { computed } from 'vue'
const props = withDefaults(defineProps<{ view: 'success' | 'critical' }>(), {
    view: 'success',
})

const classes = computed(() => ['action', `action--${props.view}`])
</script>

<style scoped>
.action {
    width: 24px;
    height: 24px;
    border-radius: 4px;
}

.action--success {
    background-color: green;
}
.action--critical {
    background-color: red;
}
</style>
