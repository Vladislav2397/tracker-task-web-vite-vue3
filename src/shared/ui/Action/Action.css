.action {
    width: 18px;
    height: 18px;
    border-radius: 4px;
}

.action--success {
    background-color: green;
}
.action--critical {
    background-color: red;
}

@media (min-width: 768px) {
    .action {
        width: 24px;
        height: 24px;
        border-radius: 4px;
    }
}