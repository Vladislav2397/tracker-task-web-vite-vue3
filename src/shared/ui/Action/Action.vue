<template>
    <div :class="classes" data-testid="Action"></div>
</template>

<script lang="ts" setup>
import { computed } from 'vue'
const props = withDefaults(defineProps<{ view: 'success' | 'critical' }>(), {
    view: 'success',
})

const classes = computed(() => ['action', `action--${props.view}`])
</script>

<style scoped>
.action {
    width: 18px;
    height: 18px;
    border-radius: 4px;
}

.action--success {
    background-color: green;
}
.action--critical {
    background-color: red;
}

@media (min-width: 768px) {
    .action {
        width: 24px;
        height: 24px;
        border-radius: 4px;
    }
}
</style>
