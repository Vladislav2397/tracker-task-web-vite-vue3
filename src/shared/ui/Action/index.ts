import Action from './Action.vue'

export { Action }