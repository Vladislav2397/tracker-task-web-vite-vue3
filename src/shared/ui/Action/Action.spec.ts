import { render } from '@testing-library/vue'
import Action from './Action.vue'

describe('shared/ui/Action component', () => {
    it.each(['success', 'critical'])('Snapshot for view %s', (view) => {
        const { html } = render(Action, {
            props: {
                view,
            },
        })

        expect(html()).toMatchSnapshot()
    })
})
