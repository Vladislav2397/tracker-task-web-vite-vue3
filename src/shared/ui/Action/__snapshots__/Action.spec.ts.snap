// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`shared/ui/Action component > Snapshot for view critical 1`] = `"<div data-v-7c2afad0=\\"\\" class=\\"action action--critical\\" data-testid=\\"Action\\"></div>"`;

exports[`shared/ui/Action component > Snapshot for view success 1`] = `"<div data-v-7c2afad0=\\"\\" class=\\"action action--success\\" data-testid=\\"Action\\"></div>"`;
